# PARTIE 1 


Le fichier titanic_train.csv est une base de données consultable avec Excel, il contient plusieurs informations notamment sur les personnes qui ont embarqué dans le Titanic. On trouve également d’autres informations dans cette base de données.

« PassengerID » qui indique le numéro du passager.
« Survived » qui indique si la personne a survécu ou non.
« Pclass » qui indique la classe du ticket.
« Name » qui indique le nom et le prénom de la personne.
« Sex » qui indique le sexe de la personne.
« Age » qui indique l’âge de la personne.
« SibSP » qui indique le nombre de frères, sœurs et d’époux cumulés.
« Parch » qui indique le nombre de parents ou d’enfants.
« Ticket » qui indique le numéro du ticket.
« Fare » qui indique le prix du ticket.
« Cabin » qui indique le numéro de la cabine.
« Embarked » qui indique le port d’embarquement.
L’objectif de cette base de données est de renseigner les informations sur les personnes ayant embarqué dans le Titanic et d’avoir des informations sur ces personnes et ainsi les retrouver plus facilement dans la base de données pour savoir par exemple si la personne est décédée ou non, cela permettrait également de savoir l’âge moyen des morts lors du naufrage du Titanic.

La base de données titanic_train.csv possède en cardinalité 891 tuples et il possède 12 en degrés. Le domaine des attributs est :

Domaine PassengerID {1 à 891}
Domaine Survived {0 ou 1}
Domaine Pclass {1,2,3}
Domaine Name (Nom prénom)
Domaine Sex {male, female}
Domaine Age (entier positif ou nul)
Domaine Sibsp (entier positif)
Domaine Parch (entier)
Domaine Ticket {A/5 21171, PC 17599, 370376}
Domaine Fare (nombre_positif)
Domaine Cabin {null, C85, C125, … , C148}
Domaine Embarked {Q, S, C}



# Phase II: Mini Rapport + Script

j ai  créer précisément 3 tables (relations).  Dans la première Table que nous avons nommée "🔺BILLETS🔺”:  nous avons 5 attributs. Le premier attribut est “Name” avec le type varchar(chaîne de caractères)  qui est elle une clé primaire car elle n'a ni emplac
ement vide (not null) ni de tuples répétitives (c'est à dire qu'il y'a que des tuples uniques).Cette attribut liste tous les prénoms des passagers. Le second attribut est "CABINE” avec un type varchar(10).Cette attribut correspond au sexe des passagers si il est 'male' ou 'female.Le troisième attribut est “FARE", il est de type float(c'est à dire un nombre àvirgule). Le quatrième attribut est “Pclass” de type integer(valeur entière) de   type int(variant de 1 à 3) qui quant à lui  indique le niveau de classe du passager en fonction de leurs moyens.Ici, cet attribut indique le nombre d'époux/femmes parmi les passagers.Le cinquième attribut est Embarked de type varchar  

Dans la deuxième table que nous avons  nommée 🔺"clients”🔺: Nous avons aussi 5 attributs. Le premier   attribut est “Passengerld” de type integer,  cet attribut est aussi la clé primaire (elle a  aucun emplacement vide et tous les tuples sont uniques).Cette attribut associe une   valeur entière propre à chaque  passager.   Le second attribut est “Sibsp” de   type int. Le troisième  attribut est “Age” de type varchar qui  correspond a l age  du passager. Le quatrième attribut est “sex” de type varchar Cet attribut montre le sex des passagers  Le dernier attribut est “parch ” de type integer et indique le nb de parents ou enfants passagers 

 Pour finir, la dernière que nous avons  nommé 🔺”survivants”🔺 est composée de 2  attributs: Le premier attribut est “Passengerld” de type integer qui est lui aussi une clé  étrangère (car déjà primaire dans la relation “Trajet”). Le second  attribut est “Survived” de type integer(variants de 0 à 1).Si le chiffre correspond à 1, le passager a survécu, si le chiffre correspond à 0, il n'a pas  survécu.

In [ ]:
%load_ext sql
print("sql active")

In [ ]:
%sql postgresql://postgres:0000@localhost/postgres 
print ("connecté!")

In [ ]:
%%sql
Create Table Titanic_train(
PassengerId integer primary key,
Survived integer,
Pclass integer, 
Name varchar(100),
Sex varchar(10),      
Age varchar(5),      
SibSp integer,
Parch integer, 
Ticket varchar(20),
Fare float,    
Cabin varchar(20),
Embarked varchar(1));

In [ ]:
%%sql
\copy titanic_train from 'C:\Users\33650\Desktop\titanic_train.csv' delimiter ',' CSV header;

In [ ]:
%%sql
select* from titanic_train

In [ ]:
%%sql
create table billets( 
Pclass integer,
Name varchar(100) primary key,
Ticket varchar(20),
Fare float,
Cabin  varchar(20),
Embarked varchar(1)
);



create table clients( 
passengerID integer primary key,
Name varchar(100),
Sex varchar(10),
Age varchar(5),
SibSp integer,
Parch integer
);


create table  survivants(
passengerID integer,
survived integer,
primary key(passengerID),  
foreign key (passengerID) references Clients (passengerID)
);



In [ ]:
%%sql
insert into billets (Pclass,Name,Ticket, Fare,Cabin,Embarked)
select Pclass,Name,Ticket,Fare,Cabin,Embarked from titanic_train;

insert into clients (PassengerID,Name,Sex,Age,SibSp,Parch) 
select PassengerID,Name,Sex,Age,SibSp,Parch from titanic_train;


insert into survivants(PassengerID,Survived)
select  PassengerID,Survived from titanic_train


In [ ]:
resul_client=%sql select * from clients
result_billet=%sql select * from billets
result_srvivant=%sql select * from survivants

In [10]:
print(resul_client)
print(result_billet)
print(result_srvivant)

None
None
None


In [ ]:
%%sql 
select * from billets


In [ ]:
%%sql
select * from clients

In [ ]:
%%sql
select * from survivants

# PARTIE 3 ##

# Questions 1 : Combien de classes de passagers différentes y avait-il à bord du Titanic ?

In [9]:
%%sql
select distinct Pclass from titanic_train;

 * postgresql://postgres:***@localhost/postgres
3 rows affected.


[(1,), (3,), (2,)]

# Questions 2 : Combien de passagers y avait-il dans chaque classe ?


In [10]:
%%sql
select count(Pclass) from titanic_train where Pclass=1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(216,)]

In [11]:
%%sql
select count(Pclass) from titanic_train where Pclass=2;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(184,)]

In [12]:
%%sql
select count(Pclass) as count_class from titanic_train where Pclass=3;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(491,)]

# Questions 3 : Combien de femmes et d'hommes y avait-il dans chaque classe ?
Dans la class 1, il y a :

122 hommes,
94 femmes.
Dans la class 2, il y a :

108 hommes,
76 femmes.
Dans la class 3, il y a :

347 hommes,
144 femmes.
Au total il y a 314 soit 35,2% de femmes et 577 soit 64,8% d'hommes.



In [13]:
%%sql
select count(sex) as count_female_class1 from titanic_train where sex='female' and Pclass=1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(94,)]

In [14]:
%%sql
select count(sex) as count_female_class2 from titanic_train where sex='female' and Pclass=2;


 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(76,)]

In [15]:
%%sql
select count(sex)  as count_female_class3 from titanic_train where sex='female' and Pclass=3;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(144,)]

In [16]:
%%sql
select count(sex) as count_male_class1 from titanic_train where sex='male' and Pclass=1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(122,)]

In [17]:
%%sql
select count(sex) as count_male_class2 from titanic_train where sex='male' and Pclass=2;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(108,)]

In [18]:
%%sql
select count(sex) as count_male_class3 from titanic_train where sex='male' and Pclass=3;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(347,)]

# Questions 4 : Comptez le nombre et le pourcentage de survivants et de passagers morts.
On a 342 survivants  et 549 morts 

In [19]:
%%sql
select count(survived) as count_survived from titanic_train where survived= 1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(342,)]

In [20]:
%%sql
select count(survived) as count_mort from titanic_train where survived= 0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(549,)]

In [21]:
#print('le pourcnetage de mort est :' ,549*100/891,'%');
#print('le pourcnetage de vivant est :' ,342*100/891,'%');



In [22]:
%%sql 
Select((count(survived)*100.00) )/ (select count(*) from survivants) as "pourcentage des morts % "
from survivants
where survived=0

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(Decimal('61.6161616161616162'),)]

In [23]:
 %%sql
select((count(survived)*100.00) )/ (select count(*) from survivants) as "pourcentage des vivants % "
from survivants
where survived=1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(Decimal('38.3838383838383838'),)]

# Questions 5 : Visualiser la répartition des passagers survivants et morts par classe.

In [ ]:
%%sql
create view  survived_classe1 as(
select count(survived) as count_survived_class1  from survivants
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=1 and survived=1 )

In [24]:
%%sql
select * from survived_classe1

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(136,)]

In [ ]:
%%sql
create view  survived_classe2 as(
select count(survived) as count_survived_class2  from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=2  and survived=1)

In [25]:
%%sql
select * from survived_classe2

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(87,)]

In [ ]:
%%sql
create view  survived_classe3 as(
select count(survived) as count_survived_class3  from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=3  and survived=1)

In [26]:
%%sql
select * from survived_classe3

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(119,)]

In [ ]:
%%sql
create view  mort_classe1 as(
select count(survived) as count_mort_class1  from survivants
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=1 and survived=0)

In [27]:
 %%sql
select * from mort_classe1 

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(80,)]

In [ ]:
%%sql
create view  mort_classe2 as(
select count(survived) as count_mort_class2  from survivants
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=2 and survived=0)

In [28]:
 %%sql
select * from mort_classe2

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(97,)]

In [ ]:
%%sql
create view  mort_classe3 as(
select count(survived) as count_mort_class3  from survivants
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where pclass=3 and survived=0)

In [527]:
 %%sql
select * from mort_classe3

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(372,)]

In [ ]:
%%sql
create view survived_homme as(
select count(survived) as count_survived_homme from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where sex='male'  and survived=1)

In [29]:
%%sql
select * from survived_homme

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(109,)]

In [ ]:
%%sql
create view survived_femme as(
select count(survived) as count_survived_femme from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where sex='female' and survived=1)

# Questions 6 : Visualiser la répartition des passagers survivants et des passagers décédés selon le sexe.

In [30]:
%%sql
select * from survived_femme

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(233,)]

In [ ]:
%%sql
create view mort_homme as(
select count(survived) as count_mort_homme from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where sex='male'  and survived=0)

In [31]:
%%sql
select * from  mort_homme

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(468,)]

In [ ]:
%%sql
create view mort_femme as(
select count(survived) as count_mort_femme from survivants 
left join  clients on survivants.passengerid= clients.passengerid 
left  join billets on billets.name=clients.name where sex='female'  and survived=0)

In [539]:
%%sql
select * from  mort_femme

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(81,)]

# ANALYSE PARTIE 3 :

Dans cette partie de exercice, on a traité  et interrogé les données de la base principale titanic_train.
Dans la première question on a extrait le nombre de classe et le nombre total des voyageurs pour chaque classe
puis à l'aide d'une requête select count () sur la table  titanic_train ,et un distinct sur  le champ Pclass pour faire sortir
que les non doublons on trouve que la titanic contient 3 classes de passager.
pour un totale de  3 passagers on trouve  que :
la classe 1 contient : 216 passagers
la classe 2 contient : 184 passagers
la classe 3 contient : 491 passagers
    
Dans la question qui suit, on a intégrrogé notre base pour savoir le nombre de femmes et hommes dans chaque classe,
on a utilisé une requete select sur la table titanic_train avec une projection sur le champs sexe et on a trouvé:
la classe 1 contient : 122 Hommes et 94 femmes
la classe 2 contient : 108 Hommes et 76 femmes.
la classe 3 contient : 347  hommes et 144 femmes.

En  ce qui suit, on a calculé le pourcentage de vivants et de morts  pour les passager avec la règles mathématique suivante :
(nombre totale ( mort ou vivants )* 100) / total des passagers
on a trouvé que il y avait  61,61 % mort et 38.38 % vivants 

Après avec les Vue, on a fait une jointure gauche entre les table client et billets  et survivants pour visualiser 
le nombre de vivants et morts selon la classe :
on a trouvé les résultats suivant :
pour la classe 1 :
                  -le nombre de mort : 80
                  -le nombre de vivant: 136
pour la classe 2 :
                  -le nombre de mort : 97
                  -le nombre de vivant: 87
pour la classe 3 :
                  -le nombre de mort : 372
                  -le nombre de vivant: 119
                  
Et enfin, toujours avec les vues, on affiche le nombre de survivant et mort  selon leur sexe:
Pour les femme:
        - nombre de surivants :233
        - nombre de morts :81
Pour les hommes :
        - nombre de survivants :109
        - nombre de morts :468